pluginsdk/support/bundlepublisherutil/bputil.go

```diff
@@ -0,0 +1,257 @@
+	spiffeBytes []byte
+}
+
+// KeyType represents the types of keys that are supported by the KeyManager.
+type BundleFormat int
+
+// BundleFormatFromString returns the BundleFormat corresponding to the provided
+// string.
+func BundleFormatFromString(s string) (BundleFormat, error) {
+	switch strings.ToLower(s) {
+	case "spiffe":
+		return SPIFFE, nil
+	case "jwks":
+		return JWKS, nil
+	case "pem":
+		return PEM, nil
+	default:
+		return BundleFormatUnset, fmt.Errorf("unknown bundle format: %q", s)
+	}
+}
+
+// NewBundle return a new *Bundle with the *types.Bundle provided.
+// Use the Bytes() function to get a slice of bytes with the bundle formatted in
+// the format specified.
+func NewBundle(pluginBundle *types.Bundle) *Bundle {
+	return &Bundle{
+		bundle: pluginBundle,
+	}
+}
+
+// String returns the string name for the bundle format.
+func (bundleFormat BundleFormat) String() string {
+	switch bundleFormat {
+	case BundleFormatUnset:
+		return "UNSET"
+	case SPIFFE:
+		return "spiffe"
+	case PEM:
+		return "pem"
+	case JWKS:
+		return "jwks"
+	default:
+		return fmt.Sprintf("UNKNOWN(%d)", int(bundleFormat))
+	}
+}
+
+// Bytes returns the bundle in the form of a slice of bytes in
+// the chosen format.
+func (b *Bundle) Bytes(format BundleFormat) ([]byte, error) {
+	if b.bundle == nil {
+		return nil, errors.New("missing bundle")
+	}
+
+	switch format {
+	case BundleFormatUnset:
+		return nil, errors.New("no format specified")
+	case JWKS:
+		if b.jwksBytes != nil {
+			return b.jwksBytes, nil
+		}
+		jwksBytes, err := b.toJWKS()
+		if err != nil {
+			return nil, fmt.Errorf("could not convert bundle to jwks format: %w", err)
+		}
+		b.jwksBytes = jwksBytes
+		return jwksBytes, nil
+	case PEM:
+		if b.pemBytes != nil {
+			return b.pemBytes, nil
+		}
+		pemBytes, err := b.toPEM()
+		if err != nil {
+			return nil, fmt.Errorf("could not convert bundle to pem format: %w", err)
+		}
+		b.pemBytes = pemBytes
+		return pemBytes, nil
+	case SPIFFE:
+		if b.spiffeBytes != nil {
+			return b.spiffeBytes, nil
+		}
+		spiffeBytes, err := b.toSPIFFEBundle()
+		if err != nil {
+			return nil, fmt.Errorf("could not convert bundle to spiffe format: %w", err)
+		}
+		b.spiffeBytes = spiffeBytes
+		return spiffeBytes, nil
+	default:
+		return nil, fmt.Errorf("invalid format: %q", format)
+	}
+}
+
+// toJWKS converts to JWKS the current bundle.
+func (b *Bundle) toJWKS() ([]byte, error) {
+	var jwks jose.JSONWebKeySet
+
+	x509Authorities, jwtAuthorities, err := getAuthorities(b.bundle)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, rootCA := range x509Authorities {
+		jwks.Keys = append(jwks.Keys, jose.JSONWebKey{
+			Key:          rootCA.PublicKey,
+			Certificates: []*x509.Certificate{rootCA},
+		})
+	}
+
+	for keyID, jwtSigningKey := range jwtAuthorities {
+		jwks.Keys = append(jwks.Keys, jose.JSONWebKey{
+			Key:   jwtSigningKey,
+			KeyID: keyID,
+		})
+	}
+
+	var out interface{} = jwks
+	return json.MarshalIndent(out, "", "    ")
```

Comment:
My vote would be to just do json.Marshal and leave indenting to the caller if they care.